app/src/main/java/com/example/frolic/EntrantDetailsActivity.java

```diff
@@ -72,8 +78,50 @@ public void onClick(View v) {
 
         // Set up filter dropdown
         tvFilter.setOnClickListener(this::showFilterMenu);
+
+        // Set up lottery picking
+        btnPickEntrants.setOnClickListener(v -> pickEntrants());
+    }
+
+    private void pickEntrants() {
+        db.collection("lotteries").document(eventId) // Use "lotteries" here
+                .get()
+                .addOnSuccessListener(documentSnapshot -> {
+                    if (documentSnapshot.exists()) {
+                        LotterySystem lottery = documentSnapshot.toObject(LotterySystem.class);
+
+                        if (lottery == null) {
+                            Log.e("PickEntrants", "LotterySystem is null");
+                            return;
+                        }
+
+                        // Call the lottery draw method to update the lists
+                        lottery.drawLottery();
+
+                        ArrayList<String> inviteds = lottery.getInvitedListIds();
+
+                        for(int i = 0; i < inviteds.size(); i++) {
+                            Log.d("", "" + inviteds.get(i));
+                        }
+                        // Update Firestore with the modified lists
+                        db.collection("lotteries").document(eventId) // Ensure "lotteries" here too
+                                .set(lottery)
```

Comment:
This sets the whole lottery object in firebase, which is problematic. What we should do instead is use a map function, see the toMap function in the events class, which gives firebase the data in basic types like Strings which it understands better, and is way more efficient. Creating a whole map here is unnecessary though, and instead you should just update the array that needs to be updated, in Firebase instead of reentering an entire lottery object into Firebase.